vortex-array/src/canonical.rs

```diff
@@ -307,6 +308,25 @@ pub trait IntoCanonical {
     fn into_canonical(self) -> VortexResult<Canonical>;
 }
 
+/// Encoding VTable for canonicalizing an array.
+#[allow(clippy::wrong_self_convention)]
+pub trait IntoCanonicalVTable {
+    fn into_canonical(&self, array: ArrayData) -> VortexResult<Canonical>;
+}
+
+/// Implement the [IntoCanonicalVTable] for all encodings with arrays implementing [IntoCanonical].
+impl<E: Encoding> IntoCanonicalVTable for E
+where
+    E::Array: IntoCanonical,
+    E::Array: TryFrom<ArrayData, Error = VortexError>,
+{
+    fn into_canonical(&self, data: ArrayData) -> VortexResult<Canonical> {
+        let canonical = E::Array::try_from(data.clone())?.into_canonical()?;
```

Comment:
TODO(ngates): let's see if we can get rid of Statistics trait in favor of StatsSet, therefore we can avoid this (albeit cheap) clone and just grab stats before canonicalizing.

The annoying thing (which was also true before), is that this function only gets called for `ArrayData::into_canonical`, not for e.g. `PrimitiveArray::into_canonical`.

The fix for this is the same as compute, to have an "entry-point" function called `into_canonical(...)` that will preserve the stats, and to remove the `IntoCanonical` trait forcing everyone to call the entry point.